ipcc/src/handle.rs

```diff
@@ -0,0 +1,107 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// Copyright 2023 Oxide Computer Company
+
+use std::{
+    ffi::{c_int, CStr, CString},
+    ptr,
+};
+
+use crate::IpccError;
+use crate::{ffi::*, IpccErrorInner};
+
+pub struct IpccHandle(*mut libipcc_handle_t);
+
+impl Drop for IpccHandle {
+    fn drop(&mut self) {
+        unsafe {
```

Comment:
For each unsafe block, it would be to have a comment explaining why the code is safe across FFI boundaries. An example of what I mean: https://github.com/oxidecomputer/omicron/blob/30d41911f3682e21f34ec041a651c3f206600894/sled-agent/src/swap_device.rs#L434-L448

Reply:
I added safety comments to unsafe blocks that were not calls into libipcc itself since calling any C function has to be marked unsafe. So all the usage of CString and CStr list why they are safe.  Let me know if you would like more safety comments elsewhere.